apps/frontend/core-dapp/src/stores/TokenSenderStore.ts

```diff
@@ -0,0 +1,41 @@
+import { ContractReturn, ContractStore, Factory } from 'prepo-stores'
+import { BigNumber } from 'ethers'
+import { RootStore } from './RootStore'
+import { TokenSenderAbi, TokenSenderAbi__factory } from '../../generated/typechain'
+import { SupportedContracts } from '../lib/contract.types'
+
+type GetScaledPrice = TokenSenderAbi['functions']['getScaledPrice']
+type GetScaledPriceLowerBound = TokenSenderAbi['functions']['getScaledPriceLowerBound']
+export class TokenSenderStore extends ContractStore<RootStore, SupportedContracts> {
+  constructor(rootStore: RootStore) {
+    super(rootStore, 'TOKEN_SENDER', TokenSenderAbi__factory as unknown as Factory)
+  }
+
+  private getScaledPrice(...params: Parameters<GetScaledPrice>): ContractReturn<GetScaledPrice> {
+    return this.call<GetScaledPrice>('getScaledPrice', params)
+  }
+
+  private getScaledPriceLowerBound(
+    ...params: Parameters<GetScaledPriceLowerBound>
+  ): ContractReturn<GetScaledPriceLowerBound> {
+    return this.call<GetScaledPriceLowerBound>('getScaledPriceLowerBound', params)
+  }
+
+  private get scaledPriceIfActive(): BigNumber | undefined {
```

Comment:
I think can name this to scaledPrice so you don't need to rename the variable name in calculateReward

```suggestion
  private get scaledPrice(): BigNumber | undefined {
```